System/AbstractFilePath/Common.hs

```diff
@@ -0,0 +1,855 @@
+{-# LANGUAGE CPP #-}
+-- This template expects CPP definitions for:
+--     WINDOWS
+--     POSIX
+--     FILEPATH_NAME = PosixFilePath | WindowsFilePath  | AbstractFilePath
+--     OSSTRING_NAME = PosixString   | WindowsString    | OsString
+--     WORD_NAME     = PosixChar     | WindowsChar      | OsChar
+--     WTOR          = PW            | WW               | OsChar
+--     CTOR          = PS            | WS               | OsString
+
+#ifdef WINDOWS
+module System.AbstractFilePath.Windows
+#elif defined(POSIX)
+module System.AbstractFilePath.Posix
+#else
+module System.AbstractFilePath
+#endif
+  (
+  -- * Types
+#ifdef WINDOWS
+    WindowsString
+  , WindowsChar
+  , WindowsFilePath
+#elif defined(POSIX)
+    PosixString
+  , PosixChar
+  , PosixFilePath
+#else
+    AbstractFilePath
+  , OsString
+  , OsChar
+#endif
+  -- * String construction
+#if defined(WINDOWS) || defined(POSIX)
+  , toPlatformString
+  , toPlatformStringIO
+  , bsToPlatformString
+  , pstr
```

Comment:
These two names are a bit cryptic, but I'm bad with naming.

Reply:
I think `pstr` is ok for this, because it's the name for the quasiquote, which really should be short to avoid annoyances.

`bsToPlatformString` expanded would be `byteStringToPlatformString`... which gives me headache :p